include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1778,18 +1895,20 @@ namespace internal
           }
 
         // ------------------------------ indices ------------------------------
-        std::vector<types::global_dof_index> level_dof_indices_fine_0(
+        std::vector<types::global_dof_index> level_dof_indices_coarse(
           transfer.schemes[0].n_dofs_per_cell_fine);
-        std::vector<types::global_dof_index> level_dof_indices_fine_1(
+        std::vector<types::global_dof_index> level_dof_indices_fine(
           transfer.schemes[1].n_dofs_per_cell_fine);
 
-        unsigned int cell_no_0 = 0;
-        unsigned int cell_no_1 = transfer.schemes[0].n_coarse_cells;
+        unsigned int n_coarse_cells_total = 0;
+        for (unsigned int i = 0; i < transfer.schemes.size(); i++)
+          {
+            n_coarse_cells_total += transfer.schemes[i].n_coarse_cells;
+          }
```

Comment:
Maybe a range-based for loop is more elegant here (otherwise, we would also prefer using prefix increment operators `++i` in deal.II over the postfix variant):
```suggestion
        for (const auto &scheme : transfer.schemes)
          n_coarse_cells_total += scheme.n_coarse_cells;
```